org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/ConfirmPanel.java

```diff
@@ -259,8 +259,15 @@ private void init()
 
         // IDEMPIERE-1334 start
         pnlBtnCenter = new Hlayout();
-        pnlBtnCenter.setSclass("confirm-panel-center");
+        pnlBtnCenter.setSclass("z-valign-middle confirm-panel-center");
+        pnlBtnCenter.setStyle("height: 100%");
         // IDEMPIERE-1334 end
+
+        // Invisible Spacer. Lets Checkboxes appear vertically aligned to center.  
+        // TODO: Find a better solution that does not require the Div height to be hardcoded.
+        Div alignmentFix = new Div();
+        alignmentFix.setStyle("content: \"\"; display: inline-block; vertical-align: middle; height: 30px;");
+        pnlBtnCenter.appendChild(alignmentFix);
```

Comment:
I'm sorry but we can't accept this hack to core ...

Reply:
I understand. I just added a comment to the jira ticket explaining why I still opened the pull request.

This is how it looks.
![Bildschirmfoto 2021-06-18 um 15 53 15](https://user-images.githubusercontent.com/2970124/122577566-3b425d00-d053-11eb-901e-94f88735fe69.png)
Note: There is one layout problem though with the modification:
To achieve the proper vertical center align I had to add a spacer to the center div that assures it has the same height as the left and right divs. Problem is I had to hardcode the div height which surely cannot be accepted in Production code because it would lead to problems when a theme is used that has a different button height.
Despite some extensive research on the issue I could not come up with a cleaner solution that does not require setting the div height in code. I still opened the pull request in the hope that a solution can be found when more eyes look into the matter. 
I also find the added functionality of having a checkbox available in the ConfigPanel very valuable.
Without the Spacer the checkbox is always top aligned regardless of the selected css class/style:
![Bildschirmfoto 2021-06-18 um 16 11 22](https://user-images.githubusercontent.com/2970124/122577629-4d240000-d053-11eb-907b-33a38466af4b.png)
In the QuickEdit Dialog it still looks acceptable because of the 9px padding above the button row.
Without the padding however it looks really crappy to have the checkbox top aligned.

Maybe we could merge nonetheless without the spacer that ensures vertical middle alignment and add a fix for the alignment problem later when one has been found.

WDYT?

Reply:
Hackish layout solution removed. 

I might add: 
The usage of a spacer element to enforce vertical middle alignment of elements of varying height is not a hack (when it can not be guaranteed that at least one of the actually used elemets has the height of the container). 
Instead it is the only solution to the problem I am aware of.  The hack IMO is the usage of a fixed height in code and I understand that this is not acceptable. 
The problem here is that (a) the height of the parent container is often dynamically calculated and because of that can not be retrieved in advance and that (b) zk uses nested divs in the implementation of vertical middle alignment. The intermediate div that is inserted by the renderer is not accessible for styling AFAIK. Because of this the layout will only work when a fixed height is assigned to the inner div. I am not an expert in zk. So I am still hopeful that the framework addresses the issue in a way that I am not aware of. I at least could not find the solution I am afraid.